src/main/java/org/tron/keystore/WalletFile.java

```diff
@@ -0,0 +1,461 @@
+
+        public Aes128CtrKdfParams() {
+        }
+
+        public int getDklen() {
+            return dklen;
+        }
+
+        public void setDklen(int dklen) {
+            this.dklen = dklen;
+        }
+
+        public int getC() {
+            return c;
+        }
+
+        public void setC(int c) {
+            this.c = c;
+        }
+
+        public String getPrf() {
+            return prf;
+        }
+
+        public void setPrf(String prf) {
+            this.prf = prf;
+        }
+
+        public String getSalt() {
+            return salt;
+        }
+
+        public void setSalt(String salt) {
+            this.salt = salt;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof Aes128CtrKdfParams)) {
+                return false;
+            }
+            
+            Aes128CtrKdfParams that = (Aes128CtrKdfParams) o;
+            
+            if (dklen != that.dklen) {
+                return false;
+            }
+            if (c != that.c) {
+                return false;
+            }
+            if (getPrf() != null
+                    ? !getPrf().equals(that.getPrf())
+                    : that.getPrf() != null) {
+                return false;
+            }
+            return getSalt() != null
+                ? getSalt().equals(that.getSalt()) : that.getSalt() == null;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = dklen;
+            result = 31 * result + c;
+            result = 31 * result + (getPrf() != null ? getPrf().hashCode() : 0);
+            result = 31 * result + (getSalt() != null ? getSalt().hashCode() : 0);
+            return result;
+        }        
+    }
+
+    public static class ScryptKdfParams implements KdfParams {
+        private int dklen;
+        private int n;
+        private int p;
+        private int r;
+        private String salt;
+
+        public ScryptKdfParams() {
+        }
+
+        public int getDklen() {
+            return dklen;
+        }
+
+        public void setDklen(int dklen) {
+            this.dklen = dklen;
+        }
+
+        public int getN() {
+            return n;
+        }
+
+        public void setN(int n) {
+            this.n = n;
+        }
+
+        public int getP() {
+            return p;
+        }
+
+        public void setP(int p) {
+            this.p = p;
+        }
+
+        public int getR() {
+            return r;
+        }
+
+        public void setR(int r) {
+            this.r = r;
+        }
+
+        public String getSalt() {
+            return salt;
+        }
+
+        public void setSalt(String salt) {
+            this.salt = salt;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof ScryptKdfParams)) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Compare to "this.getClass()" instead. ([squid:S2162](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2162))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=938&issues=AWPK23iWkDisRwWBfuxK&open=AWPK23iWkDisRwWBfuxK)

[ISSUE_KEY]: AWPK23iWkDisRwWBfuxK